### PR TITLE
Add warning in content preview when disconnected from editor #917

### PR DESCRIPTION
**Pull request by matiasmolleja**
For Issue #917

### CODE REVIEW

File: src/OrchardCore.Modules/Orchard.ContentPreview/Views/Preview/Index.cshtml [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
-            localStorage.removeItem('contentpreview:@previewId');
+            if (ev.originalEvent.key == 'contentpreview:@previewId') {
+                $(notConnectedWarning).hide();
+                renderPreview(ev.originalEvent.newValue);
+                localStorage.removeItem('contentpreview:@previewId');
+            }
+            else if (ev.originalEvent.key == 'contentpreview:not-connected:@previewId') {
+                $(notConnectedWarning).show();
+            }
+            else {
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentPreview/Views/Preview/Index.cshtml]: Unnecessary

File: src/OrchardCore.Modules/Orchard.ContentPreview/Views/Preview/Index.cshtml [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
 <script asp-name="jquery" at="Foot"></script>
 <script asp-src="/Orchard.Templates/js.cookie.js" depends-on="jquery" at="Foot"></script>
 
-<iframe id="iframe" style="position:fixed; top:0px; left:0px; bottom:0px; right:0px; width:100%; height:100%; border:none; margin:0; padding:0; overflow:hidden; z-index:999999;">
+<iframe id="iframe" style="position:fixed; top:0px; left:0px; bottom:0px; right:0px; width:100%; height:100%; border:none; margin:0; padding:0; overflow:hidden; z-index:999990;">
     Your browser doesn't support iframes
 </iframe>
 
+<div id="notConnectedWarning" style="position:fixed; right:0px; top:0px; padding:10px; margin:20px; background-color:orange; opacity:0.7; border:1px solid red; z-index:999999; display:none;">
+    <p style="margin:0; padding:5px 10px; font-family:sans-serif; font-size:0.8em; color:red; border:1px dashed red;">preview not connected</p>
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentPreview/Views/Preview/Index.cshtml]: Text needs to be localized

File: src/OrchardCore.Modules/Orchard.ContentPreview/Views/Preview/Index.cshtml [hunk 3]
```diff
@@ -1,25 +1,38 @@
...
 
 <script asp-name="jquery" at="Foot"></script>
 <script asp-src="/Orchard.Templates/js.cookie.js" depends-on="jquery" at="Foot"></script>
 
-<iframe id="iframe" style="position:fixed; top:0px; left:0px; bottom:0px; right:0px; width:100%; height:100%; border:none; margin:0; padding:0; overflow:hidden; z-index:999999;">
+<iframe id="iframe" style="position:fixed; top:0px; left:0px; bottom:0px; right:0px; width:100%; height:100%; border:none; margin:0; padding:0; overflow:hidden; z-index:999990;">
     Your browser doesn't support iframes
 </iframe>
 
+<div id="notConnectedWarning" style="position:fixed; right:0px; top:0px; padding:10px; margin:20px; background-color:orange; opacity:0.7; border:1px solid red; z-index:999999; display:none;">
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentPreview/Views/Preview/Index.cshtml]: Maybe we should link to bootstrap and use a standard alert.

File: src/OrchardCore.Modules/Orchard.ContentPreview/Content/Scripts/contentpreview.edit.js [hunk 4]
```diff
@@ -68,4 +68,9 @@ $(function () {
         // to pre-render the view even if no contentpreview:render is already sent
         sendFormData();
     });    
+
+    $(window).on('unload', function(){
+        // this will raise an event in the preview window to notify that the live preview is no longer active.
+        localStorage.setItem('contentpreview:not-connected:' + previewId, '');
```
Comment (sebastienros) [inline src/OrchardCore.Modules/Orchard.ContentPreview/Content/Scripts/contentpreview.edit.js]: is there a way to send events across windows without local storage in this case? Like using `jQuery $(window).trigger(...)` ?

Comment (matiasmolleja) [inline src/OrchardCore.Modules/Orchard.ContentPreview/Content/Scripts/contentpreview.edit.js]: I've been reading and I think that the only way would be by opening the preview through javascript. Something like this: 
`
var childWindow = window.open(urlToPreview);
$(childWindow).trigger( some event in the child);
`
And from the child-preview we could access the parent-admin through the 'window.opener" property that is automatically populated.

I found this post useful: [Easy Inter-window Communications with jQuery](https://objectpartners.com/2012/07/31/easy-inter-window-communications-with-jquery/)
Right now we use an anchor tag to open the preview window.
Do you think it is a good idea to mix both approaches to cross-window communication ( jquery.trigger vs storage  event) ?
The storage event way seems to work fine. But if you think it is better to use the $(window).trigger approach I can investigate further.

**Comment by matiasmolleja** [comment]
Sorry. 
I broke the PR when rebasing to dev. 
I think it is easier to simply create a new one. The new one is #1315.
I'm closing this one.

**Pull request closed without merging**